src/uct/ib/base/ib_md.c

```diff
@@ -432,8 +445,10 @@ uct_ib_md_handle_mr_list_mt(uct_ib_md_t *md, void *address, size_t length,
 
     if (status != UCS_OK) {
         for (mr_idx = 0; mr_idx < mr_num; mr_idx++) {
-            /* coverity[check_return] */
-            uct_ib_dereg_mr(mrs[mr_idx]);
+            if (mrs[mr_idx] != NULL) {
+                /* coverity[check_return] */
+                uct_ib_dereg_mr(mrs[mr_idx]);
+            }
```

Comment:
maybe add `mrs[mr_idx] = NULL`?

Reply:
not sure, because if we return an error in this function, all `mrs` are considered invalid anyways.

Reply:
IMO better to not initialize all mrs to NULL when registration fails, but instead move the cleanup loop from here to lines 438-440 flow - if one thread failed, cleanup all the other threads. We can do this easier by extracting the deregistration loop to a helper function.